### DISCUSSION TITLE
Use gingo as e2e test framework

### DISCUSSION THREAD

**Discussion by yxxhero** [helmfile/helmfile#15] [Ideas]
https://github.com/onsi/ginkgo
Use gingo as e2e test framework

**Comment by mumoshu** [comment]
Why? I use it in several projects but I'm not particularly fond of it...

**Reply by yxxhero** [reply]
1. Many open source projects are in use，ex: k8s
2. Any other recommendations?

**Reply by mumoshu** [reply]
I was wondering if we could just use `go test`. Because it's well maintained, work without additional dependencies, many open source projects use it, famous code editors and IDEs like VS Code and IDEA supports analyzing go-test code to allow running a single sub-test which is very helpful when you need to tweak a test.

**Reply by yxxhero** [reply]
ok. I agree with you. *go test* is easy.
